code/datums/components/material_container.dm

```diff
@@ -448,3 +456,36 @@
 		))
 
 	return data
+
+/**
+ * Adds context sensitivy directly to the material container file for screentips
+ * Arguments:
+ * * source - refers to item that will display its screentip
+ * * context - refers to, in this case, an item in the users hand hovering over the material container, such as an autolathe
+ * * held_item - refers to the item that has materials accepted by the material container
+ * * user - refers to user who will see the screentip when the proper context and tool are there
+ */
+/datum/component/material_container/proc/on_requesting_context_from_item(datum/source, list/context, obj/item/held_item, mob/living/user)
+	SIGNAL_HANDLER
+
+	if(isnull(held_item))
+		return NONE
+	if(!(mat_container_flags & MATCONTAINER_ANY_INTENT) && user.combat_mode)
+		return NONE
+	if(held_item.item_flags & ABSTRACT)
+		return NONE
+	if((held_item.flags_1 & HOLOGRAM_1) || (held_item.item_flags & NO_MAT_REDEMPTION) || (allowed_item_typecache && !is_type_in_typecache(held_item, allowed_item_typecache)))
+		return NONE
+	var/list/item_materials = held_item.get_material_composition(mat_container_flags)
+	var/has_materials = FALSE
+	for(var/material in item_materials)
+		if(!can_hold_material(material))
+			continue
+		has_materials = TRUE
+		break
+	if(!has_materials)
+		return NONE
```

Comment:
```suggestion
	for(var/material in item_materials)
		if(can_hold_material(material))
			continue
		return NONE
```

We can just return out of the for loop directly, can't we?

Reply:
i had to also add a line to make it quit out if there werent any materials because otherwise it would lie to you about being able to put gloves in the lathe